### ISSUE TITLE
HTTP ERROR: 503

### ISSUE THREAD

**Issue by ILiuDongHui** [gitblit/gitblit#1315]
# giblit.cmd output
2019-08-29 10:32:26 [INFO ]
  _____  _  _    _      _  _  _
 |  __ \(_)| |  | |    | |(_)| |
 | |  \/ _ | |_ | |__  | | _ | |_
 | | __ | || __|| '_ \ | || || __|  http://gitblit.com
 | |_\ \| || |_ | |_) || || || |_   @gitblit
  \____/|_| \__||_.__/ |_||_| \__|  1.8.0

2019-08-29 10:32:26 [INFO ] Running on Windows 10 (10.0)
2019-08-29 10:32:26 [INFO ] Logging initialized @394ms
2019-08-29 10:32:26 [INFO ] Using JCE Unlimited Strength Jurisdiction Policy files
2019-08-29 10:32:26 [INFO ] Setting up HTTPS transport on port 8443
2019-08-29 10:32:26 [INFO ]    certificate alias = localhost
2019-08-29 10:32:26 [INFO ]    keyStorePath   = F:\Users\Shine\Downloads\gitblit-1.8.0\data\serverKeyStore.jks
2019-08-29 10:32:26 [INFO ]    trustStorePath = F:\Users\Shine\Downloads\gitblit-1.8.0\data\serverTrustStore.jks
2019-08-29 10:32:26 [INFO ]    crlPath        = F:\Users\Shine\Downloads\gitblit-1.8.0\data\certs\caRevocationList.crl
2019-08-29 10:32:26 [WARN ] Binding HTTPS transport on port 8443 to localhost
2019-08-29 10:32:26 [INFO ] Setting up HTTP transport on port 10101
2019-08-29 10:32:26 [WARN ] Binding HTTP transport on port 10101 to 192.168.1.36
2019-08-29 10:32:27 [INFO ] Shutdown Monitor listening on port 8081
2019-08-29 10:32:27 [INFO ] jetty-9.2.13.v20150730
2019-08-29 10:32:34 [INFO ] NO JSP Support for /, did not find org.eclipse.jetty.jsp.JettyJspServlet
WARNING: An illegal reflective access operation has occurred
WARNING: Illegal reflective access by com.google.inject.internal.cglib.core.$ReflectUtils$2 (file:/F:/Users/Shine/Downloads/gitblit-1.8.0/ext/guice-4.0.jar) to method java.lang.ClassLoader.defineClass(java.lang.String,byte[],int,int,java.security.ProtectionDomain)
WARNING: Please consider reporting this to the maintainers of com.google.inject.internal.cglib.core.$ReflectUtils$2
WARNING: Use --illegal-access=warn to enable warnings of further illegal reflective access operations
WARNING: All illegal access operations will be denied in a future release
2019-08-29 10:32:34 [INFO ]
2019-08-29 10:32:34 [INFO ] ----[com.gitblit.manager.IRuntimeManager]----
2019-08-29 10:32:34 [INFO ] Basefolder  : F:\Users\Shine\Downloads\gitblit-1.8.0\data
2019-08-29 10:32:34 [INFO ] Settings    : F:\Users\Shine\Downloads\gitblit-1.8.0\data\gitblit.properties
2019-08-29 10:32:34 [INFO ] JVM timezone: Asia/Shanghai (CST +0800)
2019-08-29 10:32:34 [INFO ] App timezone: Asia/Shanghai (CST +0800)
2019-08-29 10:32:34 [INFO ] JVM locale  : zh_CN
2019-08-29 10:32:34 [INFO ] App locale  : <client>
2019-08-29 10:32:34 [WARN ] Failed startup of context o.e.j.w.WebAppContext@4c398c80{/,file:/F:/Users/Shine/Downloads/gitblit-1.8.0/data/temp/webapp/,STARTING}{file:/F:/Users/Shine/Downloads/gitblit-1.8.0/gitblit.jar}
com.google.inject.ProvisionException: Unable to provision, see the following errors:

1) Error injecting constructor, java.lang.NoClassDefFoundError: javax/activation/DataSource
  at com.gitblit.manager.NotificationManager.<init>(NotificationManager.java:55)
  at com.gitblit.manager.NotificationManager.class(NotificationManager.java:43)
  while locating com.gitblit.manager.NotificationManager
  while locating com.gitblit.manager.INotificationManager

1 error
        at com.google.inject.internal.InjectorImpl$2.get(InjectorImpl.java:1025)
        at com.google.inject.internal.InjectorImpl.getInstance(InjectorImpl.java:1051)
        at com.gitblit.servlet.GitblitContext.loadManager(GitblitContext.java:260)
        at com.gitblit.servlet.GitblitContext.startManager(GitblitContext.java:265)
        at com.gitblit.servlet.GitblitContext.startCore(GitblitContext.java:199)
        at com.gitblit.servlet.GitblitContext.contextInitialized(GitblitContext.java:148)
        at org.eclipse.jetty.server.handler.ContextHandler.callContextInitialized(ContextHandler.java:800)
        at org.eclipse.jetty.servlet.ServletContextHandler.callContextInitialized(ServletContextHandler.java:444)
        at org.eclipse.jetty.server.handler.ContextHandler.startContext(ContextHandler.java:791)
        at org.eclipse.jetty.servlet.ServletContextHandler.startContext(ServletContextHandler.java:294)
        at org.eclipse.jetty.webapp.WebAppContext.startWebapp(WebAppContext.java:1349)
        at org.eclipse.jetty.webapp.WebAppContext.startContext(WebAppContext.java:1342)
        at org.eclipse.jetty.server.handler.ContextHandler.doStart(ContextHandler.java:741)
        at org.eclipse.jetty.webapp.WebAppContext.doStart(WebAppContext.java:505)
        at org.eclipse.jetty.util.component.AbstractLifeCycle.start(AbstractLifeCycle.java:68)
        at org.eclipse.jetty.util.component.ContainerLifeCycle.start(ContainerLifeCycle.java:132)
        at org.eclipse.jetty.server.Server.start(Server.java:387)
        at org.eclipse.jetty.util.component.ContainerLifeCycle.doStart(ContainerLifeCycle.java:114)
        at org.eclipse.jetty.server.handler.AbstractHandler.doStart(AbstractHandler.java:61)
        at org.eclipse.jetty.server.Server.doStart(Server.java:354)
        at org.eclipse.jetty.util.component.AbstractLifeCycle.start(AbstractLifeCycle.java:68)
        at com.gitblit.GitBlitServer.start(GitBlitServer.java:460)
        at com.gitblit.GitBlitServer.main(GitBlitServer.java:124)
Caused by: java.lang.NoClassDefFoundError: javax/activation/DataSource
        at com.gitblit.manager.NotificationManager.<init>(NotificationManager.java:57)
        at com.gitblit.manager.NotificationManager$$FastClassByGuice$$28e6c2cd.newInstance(<generated>)
        at com.google.inject.internal.cglib.reflect.$FastConstructor.newInstance(FastConstructor.java:40)
        at com.google.inject.internal.DefaultConstructionProxyFactory$1.newInstance(DefaultConstructionProxyFactory.java:61)
        at com.google.inject.internal.ConstructorInjector.provision(ConstructorInjector.java:105)
        at com.google.inject.internal.ConstructorInjector.construct(ConstructorInjector.java:85)
        at com.google.inject.internal.ConstructorBindingImpl$Factory.get(ConstructorBindingImpl.java:267)
        at com.google.inject.internal.ProviderToInternalFactoryAdapter$1.call(ProviderToInternalFactoryAdapter.java:46)
        at com.google.inject.internal.InjectorImpl.callInContext(InjectorImpl.java:1103)
        at com.google.inject.internal.ProviderToInternalFactoryAdapter.get(ProviderToInternalFactoryAdapter.java:40)
        at com.google.inject.internal.SingletonScope$1.get(SingletonScope.java:145)
        at com.google.inject.internal.InternalFactoryToProviderAdapter.get(InternalFactoryToProviderAdapter.java:41)
        at com.google.inject.internal.FactoryProxy.get(FactoryProxy.java:56)
        at com.google.inject.internal.InjectorImpl$2$1.call(InjectorImpl.java:1016)
        at com.google.inject.internal.InjectorImpl.callInContext(InjectorImpl.java:1092)
        at com.google.inject.internal.InjectorImpl$2.get(InjectorImpl.java:1012)
        ... 22 more
Caused by: java.lang.ClassNotFoundException: javax.activation.DataSource
        at java.base/jdk.internal.loader.BuiltinClassLoader.loadClass(BuiltinClassLoader.java:583)
        at java.base/jdk.internal.loader.ClassLoaders$AppClassLoader.loadClass(ClassLoaders.java:178)
        at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:521)
        ... 38 more
2019-08-29 10:32:34 [INFO ] Started ServerConnector@302b34a0{SSL-HTTP/1.1}{localhost:8443}
2019-08-29 10:32:34 [INFO ] Started ServerConnector@51850751{HTTP/1.1}{192.168.1.36:10101}
2019-08-29 10:32:34 [INFO ] Started @8562ms

# Error
**When visiting http://192.168.1.36, I get an error：**

HTTP ERROR: 503
Problem accessing /. Reason:

    Service Unavailable

# My JDK
The JDK is OpenJDK 12

# My gitblit.cmd
`@java -cp gitblit.jar;"%CD%\ext\*" com.gitblit.GitBlitServer --baseFolder data %*`

**Comment by ILiuDongHui** [comment]
gitblit-1.8.0.zip（GitBlit GO Windows）

**Comment by ILiuDongHui** [comment]
I hope to get some help. thanks

**Comment by ybosy** [comment]
Gitblit requires a Java 7 Runtime Environment (JRE) or a Java 7 Development Kit (JDK).

**Comment by ILiuDongHui** [comment]
> Gitblit requires a Java 7 Runtime Environment (JRE) or a Java 7 Development Kit (JDK).

My question has been solved, but i am still very grateful for your kind help, thank you!

**Comment by x1244** [comment]
How to solve this  @problem.
I meet the same problem. 
Must use Java 7 Runtime Environment (JRE) or a Java 7 Development Kit (JDK)?
wait for you reply.

> > Gitblit requires a Java 7 Runtime Environment (JRE) or a Java 7 Development Kit (JDK).
> 
> My question has been solved, but i am still very grateful for your kind help, thank you!

**Comment by flaix** [comment]
This has been solved in version 1.9. Gitblit 1.9 is still built for Java 7, but it also runs fine with Java 8 and Java 11.

What problem exactly are you having and what is your environment?

**Comment by x1244** [comment]
My java SDK is v16.

```
> java --version
java 16.0.1 2021-04-20
Java(TM) SE Runtime Environment (build 16.0.1+9-24)
Java HotSpot(TM) 64-Bit Server VM (build 16.0.1+9-24, mixed mode, sharing)
```

When start gitblit,log is

```
[2021-07-27 08:09:15] [info]  Stopping service...
[2021-07-27 08:09:16] [info]  Service stopped.
[2021-07-27 08:09:16] [info]  Run service finished.
[2021-07-27 08:09:16] [info]  Commons Daemon procrun finished
[2021-07-27 08:09:19] [info]  Commons Daemon procrun (1.0.10.0 64-bit) started
[2021-07-27 08:09:19] [info]  Running 'gitblit' Service...
[2021-07-27 08:09:19] [info]  Starting service...
[2021-07-27 08:09:20] [info]  Service started in 1295 ms.
```

stderr

```
2021-07-27 08:09:19 Commons Daemon procrun stderr initialized
```

stdout

```
2021-07-27 08:09:19 Commons Daemon procrun stdout initialized
2021-07-27 08:09:19 [INFO ] 
  _____  _  _    _      _  _  _
 |  __ \(_)| |  | |    | |(_)| |
 | |  \/ _ | |_ | |__  | | _ | |_
 | | __ | || __|| '_ \ | || || __|  http://gitblit.com
 | |_\ \| || |_ | |_) || || || |_   @gitblit
  \____/|_| \__||_.__/ |_||_| \__|  1.9.1

2021-07-27 08:09:19 [INFO ] Running on Windows Server 2019 (10.0)
2021-07-27 08:09:19 [INFO ] JVM version 16.0.1 (Oracle Corporation)
2021-07-27 08:09:19 [INFO ] Logging initialized @437ms
2021-07-27 08:09:20 [INFO ] Using JCE Unlimited Strength Jurisdiction Policy files
2021-07-27 08:09:20 [INFO ] Setting up HTTPS transport on port 8443
2021-07-27 08:09:20 [INFO ]    certificate alias = localhost
2021-07-27 08:09:20 [INFO ]    keyStorePath   = C:\soft\gitblit-1.9.1\data\serverKeyStore.jks
2021-07-27 08:09:20 [INFO ]    trustStorePath = C:\soft\gitblit-1.9.1\data\serverTrustStore.jks
2021-07-27 08:09:20 [INFO ]    crlPath        = C:\soft\gitblit-1.9.1\data\certs\caRevocationList.crl
2021-07-27 08:09:20 [INFO ] Setting up HTTP transport on port 8090
2021-07-27 08:09:20 [INFO ] Shutdown Monitor listening on port 8081
2021-07-27 08:09:20 [INFO ] jetty-9.2.13.v20150730
2021-07-27 08:09:30 [INFO ] NO JSP Support for /, did not find org.eclipse.jetty.jsp.JettyJspServlet
2021-07-27 08:09:30 [WARN ] FAILED o.e.j.w.WebAppContext@45312be2{/,file:/C:/soft/gitblit-1.9.1/data/temp/webapp/,STARTING}{file:/C:/soft/gitblit-1.9.1/gitblit.jar}: java.lang.ExceptionInInitializerError
java.lang.ExceptionInInitializerError
	at com.google.inject.internal.cglib.reflect.$FastClassEmitter.<init>(FastClassEmitter.java:67)
	at com.google.inject.internal.cglib.reflect.$FastClass$Generator.generateClass(FastClass.java:72)
	at com.google.inject.internal.cglib.core.$DefaultGeneratorStrategy.generate(DefaultGeneratorStrategy.java:25)
	at com.google.inject.internal.cglib.core.$AbstractClassGenerator.create(AbstractClassGenerator.java:216)
	at com.google.inject.internal.cglib.reflect.$FastClass$Generator.create(FastClass.java:64)
	at com.google.inject.internal.BytecodeGen.newFastClass(BytecodeGen.java:204)
	at com.google.inject.internal.ProviderMethod$FastClassProviderMethod.<init>(ProviderMethod.java:256)
	at com.google.inject.internal.ProviderMethod.create(ProviderMethod.java:71)
	at com.google.inject.internal.ProviderMethodsModule.createProviderMethod(ProviderMethodsModule.java:275)
	at com.google.inject.internal.ProviderMethodsModule.getProviderMethods(ProviderMethodsModule.java:144)
	at com.google.inject.internal.ProviderMethodsModule.configure(ProviderMethodsModule.java:123)
	at com.google.inject.spi.Elements$RecordingBinder.install(Elements.java:340)
	at com.google.inject.spi.Elements$RecordingBinder.install(Elements.java:349)
	at com.google.inject.AbstractModule.install(AbstractModule.java:122)
	at com.google.inject.servlet.ServletModule.configure(ServletModule.java:52)
	at com.google.inject.AbstractModule.configure(AbstractModule.java:62)
	at com.google.inject.spi.Elements$RecordingBinder.install(Elements.java:340)
	at com.google.inject.spi.Elements.getElements(Elements.java:110)
	at com.google.inject.internal.InjectorShell$Builder.build(InjectorShell.java:138)
	at com.google.inject.internal.InternalInjectorCreator.build(InternalInjectorCreator.java:104)
	at com.google.inject.Guice.createInjector(Guice.java:96)
	at com.google.inject.Guice.createInjector(Guice.java:73)
	at com.google.inject.Guice.createInjector(Guice.java:62)
	at com.gitblit.servlet.GitblitContext.getInjector(GitblitContext.java:127)
	at com.google.inject.servlet.GuiceServletContextListener.contextInitialized(GuiceServletContextListener.java:47)
	at com.gitblit.servlet.GitblitContext.contextInitialized(GitblitContext.java:145)
	at org.eclipse.jetty.server.handler.ContextHandler.callContextInitialized(ContextHandler.java:800)
	at org.eclipse.jetty.servlet.ServletContextHandler.callContextInitialized(ServletContextHandler.java:444)
	at org.eclipse.jetty.server.handler.ContextHandler.startContext(ContextHandler.java:791)
	at org.eclipse.jetty.servlet.ServletContextHandler.startContext(ServletContextHandler.java:294)
	at org.eclipse.jetty.webapp.WebAppContext.startWebapp(WebAppContext.java:1349)
	at org.eclipse.jetty.webapp.WebAppContext.startContext(WebAppContext.java:1342)
	at org.eclipse.jetty.server.handler.ContextHandler.doStart(ContextHandler.java:741)
	at org.eclipse.jetty.webapp.WebAppContext.doStart(WebAppContext.java:505)
	at org.eclipse.jetty.util.component.AbstractLifeCycle.start(AbstractLifeCycle.java:68)
	at org.eclipse.jetty.util.component.ContainerLifeCycle.start(ContainerLifeCycle.java:132)
	at org.eclipse.jetty.server.Server.start(Server.java:387)
	at org.eclipse.jetty.util.component.ContainerLifeCycle.doStart(ContainerLifeCycle.java:114)
	at org.eclipse.jetty.server.handler.AbstractHandler.doStart(AbstractHandler.java:61)
	at org.eclipse.jetty.server.Server.doStart(Server.java:354)
	at org.eclipse.jetty.util.component.AbstractLifeCycle.start(AbstractLifeCycle.java:68)
	at com.gitblit.GitBlitServer.start(GitBlitServer.java:465)
	at com.gitblit.GitBlitServer.main(GitBlitServer.java:124)
Caused by: java.lang.reflect.InaccessibleObjectException: Unable to make protected final java.lang.Class java.lang.ClassLoader.defineClass(java.lang.String,byte[],int,int,java.security.ProtectionDomain) throws java.lang.ClassFormatError accessible: module java.base does not "opens java.lang" to unnamed module @3b084709
	at java.base/java.lang.reflect.AccessibleObject.checkCanSetAccessible(AccessibleObject.java:357)
	at java.base/java.lang.reflect.AccessibleObject.checkCanSetAccessible(AccessibleObject.java:297)
	at java.base/java.lang.reflect.Method.checkCanSetAccessible(Method.java:199)
	at java.base/java.lang.reflect.Method.setAccessible(Method.java:193)
	at com.google.inject.internal.cglib.core.$ReflectUtils$2.run(ReflectUtils.java:56)
	at java.base/java.security.AccessController.doPrivileged(AccessController.java:312)
	at com.google.inject.internal.cglib.core.$ReflectUtils.<clinit>(ReflectUtils.java:46)
	... 43 more
2021-07-27 08:09:30 [INFO ] Started ServerConnector@6ebf0f36{SSL-HTTP/1.1}{0.0.0.0:8443}
2021-07-27 08:09:30 [INFO ] Started ServerConnector@18920cc{HTTP/1.1}{0.0.0.0:8090}
2021-07-27 08:09:30 [WARN ] FAILED org.eclipse.jetty.server.Server@2807bdeb: java.lang.ExceptionInInitializerError
java.lang.ExceptionInInitializerError
	at com.google.inject.internal.cglib.reflect.$FastClassEmitter.<init>(FastClassEmitter.java:67)
	at com.google.inject.internal.cglib.reflect.$FastClass$Generator.generateClass(FastClass.java:72)
	at com.google.inject.internal.cglib.core.$DefaultGeneratorStrategy.generate(DefaultGeneratorStrategy.java:25)
	at com.google.inject.internal.cglib.core.$AbstractClassGenerator.create(AbstractClassGenerator.java:216)
	at com.google.inject.internal.cglib.reflect.$FastClass$Generator.create(FastClass.java:64)
	at com.google.inject.internal.BytecodeGen.newFastClass(BytecodeGen.java:204)
	at com.google.inject.internal.ProviderMethod$FastClassProviderMethod.<init>(ProviderMethod.java:256)
	at com.google.inject.internal.ProviderMethod.create(ProviderMethod.java:71)
	at com.google.inject.internal.ProviderMethodsModule.createProviderMethod(ProviderMethodsModule.java:275)
	at com.google.inject.internal.ProviderMethodsModule.getProviderMethods(ProviderMethodsModule.java:144)
	at com.google.inject.internal.ProviderMethodsModule.configure(ProviderMethodsModule.java:123)
	at com.google.inject.spi.Elements$RecordingBinder.install(Elements.java:340)
	at com.google.inject.spi.Elements$RecordingBinder.install(Elements.java:349)
	at com.google.inject.AbstractModule.install(AbstractModule.java:122)
	at com.google.inject.servlet.ServletModule.configure(ServletModule.java:52)
	at com.google.inject.AbstractModule.configure(AbstractModule.java:62)
	at com.google.inject.spi.Elements$RecordingBinder.install(Elements.java:340)
	at com.google.inject.spi.Elements.getElements(Elements.java:110)
	at com.google.inject.internal.InjectorShell$Builder.build(InjectorShell.java:138)
	at com.google.inject.internal.InternalInjectorCreator.build(InternalInjectorCreator.java:104)
	at com.google.inject.Guice.createInjector(Guice.java:96)
	at com.google.inject.Guice.createInjector(Guice.java:73)
	at com.google.inject.Guice.createInjector(Guice.java:62)
	at com.gitblit.servlet.GitblitContext.getInjector(GitblitContext.java:127)
	at com.google.inject.servlet.GuiceServletContextListener.contextInitialized(GuiceServletContextListener.java:47)
	at com.gitblit.servlet.GitblitContext.contextInitialized(GitblitContext.java:145)
	at org.eclipse.jetty.server.handler.ContextHandler.callContextInitialized(ContextHandler.java:800)
	at org.eclipse.jetty.servlet.ServletContextHandler.callContextInitialized(ServletContextHandler.java:444)
	at org.eclipse.jetty.server.handler.ContextHandler.startContext(ContextHandler.java:791)
	at org.eclipse.jetty.servlet.ServletContextHandler.startContext(ServletContextHandler.java:294)
	at org.eclipse.jetty.webapp.WebAppContext.startWebapp(WebAppContext.java:1349)
	at org.eclipse.jetty.webapp.WebAppContext.startContext(WebAppContext.java:1342)
	at org.eclipse.jetty.server.handler.ContextHandler.doStart(ContextHandler.java:741)
	at org.eclipse.jetty.webapp.WebAppContext.doStart(WebAppContext.java:505)
	at org.eclipse.jetty.util.component.AbstractLifeCycle.start(AbstractLifeCycle.java:68)
	at org.eclipse.jetty.util.component.ContainerLifeCycle.start(ContainerLifeCycle.java:132)
	at org.eclipse.jetty.server.Server.start(Server.java:387)
	at org.eclipse.jetty.util.component.ContainerLifeCycle.doStart(ContainerLifeCycle.java:114)
	at org.eclipse.jetty.server.handler.AbstractHandler.doStart(AbstractHandler.java:61)
	at org.eclipse.jetty.server.Server.doStart(Server.java:354)
	at org.eclipse.jetty.util.component.AbstractLifeCycle.start(AbstractLifeCycle.java:68)
	at com.gitblit.GitBlitServer.start(GitBlitServer.java:465)
	at com.gitblit.GitBlitServer.main(GitBlitServer.java:124)
Caused by: java.lang.reflect.InaccessibleObjectException: Unable to make protected final java.lang.Class java.lang.ClassLoader.defineClass(java.lang.String,byte[],int,int,java.security.ProtectionDomain) throws java.lang.ClassFormatError accessible: module java.base does not "opens java.lang" to unnamed module @3b084709
	at java.base/java.lang.reflect.AccessibleObject.checkCanSetAccessible(AccessibleObject.java:357)
	at java.base/java.lang.reflect.AccessibleObject.checkCanSetAccessible(AccessibleObject.java:297)
	at java.base/java.lang.reflect.Method.checkCanSetAccessible(Method.java:199)
	at java.base/java.lang.reflect.Method.setAccessible(Method.java:193)
	at com.google.inject.internal.cglib.core.$ReflectUtils$2.run(ReflectUtils.java:56)
	at java.base/java.security.AccessController.doPrivileged(AccessController.java:312)
	at com.google.inject.internal.cglib.core.$ReflectUtils.<clinit>(ReflectUtils.java:46)
	... 43 more
```

When access http://localhost:8090, the browser shows TIMEOUT. This should be gitblit can not working with jave v16.

So, I use gitblit war, with java v16 and Tomcat9, extract gitblit war to tomcat/webapps, it works for me.

Thanks.
